model/spectrum.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * spectrum.h
+ */
+#ifndef DXTBX_MODEL_SPECTRUM_H
+#define DXTBX_MODEL_SPECTRUM_H
+
+#include <iostream>
+#include <cmath>
+#include <scitbx/vec3.h>
+#include <scitbx/array_family/shared.h>
+#include <scitbx/array_family/simple_io.h>
+#include <scitbx/array_family/simple_tiny_io.h>
+#include <dxtbx/error.h>
+#include "model_helpers.h"
+
+namespace dxtbx { namespace model {
+  typedef scitbx::af::shared<double> vecd;
+
+  /** A class to represent a 2D spectrum. */
+  class Spectrum {
+  public:
+    /** Default constructor: initialise all to zero */
```

Comment:
While we're talking constructors, one which takes a minimum and maximum energy window, a central energy and a width to compute a Gaussian representation (see comments elsewhere) would probably be enough to stop me nagging on the topic of Polymorphism 😉

Reply:
This implementation of Spectrum uses energy channels, not bandpass.  I'm interested in this idea, but I think I need more detail and perhaps a more complete API spec.  It sounds like an expansion of the Spectrum class, not necessarily a polymorph, which could be done with a new constructor as you say.  Could that be separate work from this PR?

Reply:
More detail and API spec I can do; and as slight expansion of this class is doable => indeed no need for polymorphism. I would like to give it a chance to get the changes in here as I think in the long game it will make it more useful in one go. Min, max energy window at the very least seems like something useful to have as part of this PR, since this is what we will be using to decide what to predict on the images. 

BTW, re: min / max - was thinking the narrowest interval which contains say 98% of the total flux so it is well defined for measured and Gaussian input spectra.